File: app/src/main/java/com/example/firabasefirstexperience/activity/ResetPassActivity.java
package com.example.firabasefirstexperience.activity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.firabasefirstexperience.R;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;

public class ResetPassActivity extends AppCompatActivity {

    private EditText etEmailReset;

    private FirebaseAuth auth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reset_pass);

        etEmailReset = findViewById(R.id.etEmailReset);
        Button butReset = findViewById(R.id.butReset);


        Toolbar toolbar = findViewById(R.id.toolBarForResetPass);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle("Reset Password");
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        auth = FirebaseAuth.getInstance();

        butReset.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String eMail = etEmailReset.getText().toString();

                if (eMail.isEmpty()) {
                    Toast.makeText(ResetPassActivity.this, "Please, write your Email", Toast.LENGTH_SHORT).show();
                } else {
                    auth.sendPasswordResetEmail(eMail).addOnCompleteListener(new OnCompleteListener<Void>() {
                        @Override
                        public void onComplete(@NonNull Task<Void> task) {
                            if (task.isSuccessful()) {
                                Toast.makeText(ResetPassActivity.this, "Message with instruction send to your Email. Please Check it", Toast.LENGTH_LONG).show();
                                startActivity(new Intent(ResetPassActivity.this, LoginActivity.class));
                            } else {
                                String error = task.getException().getMessage();
                                Toast.makeText(ResetPassActivity.this, error, Toast.LENGTH_SHORT).show();
                            }
                        }
                    });
                }
            }
        });
    }
}

File: app/src/main/java/com/example/firabasefirstexperience/kotlin/activity/ResetPassActivity.kt
package com.example.firabasefirstexperience.kotlin.activity

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.firabasefirstexperience.R
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_reset_pass.*
import kotlinx.android.synthetic.main.bar_layout.*

class ResetPassActivity : AppCompatActivity() {


    private var auth: FirebaseAuth? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reset_pass)


        setSupportActionBar(tbForRegisterActivity)
        supportActionBar?.title = "Reset Password"
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        auth = FirebaseAuth.getInstance()

        butReset.setOnClickListener {
            val eMail = etEmailReset.text.toString()

            if (eMail.isEmpty()) {
                Toast.makeText(applicationContext, "Please, write your Email", Toast.LENGTH_SHORT).show()
            } else run {
                auth?.sendPasswordResetEmail(eMail)?.addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        Toast.makeText(this@ResetPassActivity, "Message with instruction send to your Email. Please Check it", Toast.LENGTH_LONG).show()
                        startActivity(Intent(this@ResetPassActivity, LoginActivity::class.java))
                    } else {
                        val error = task.exception!!.message
                        Toast.makeText(this@ResetPassActivity, error, Toast.LENGTH_SHORT).show()
                    }
                }
            }
        }


    }
}